lib/ansible/modules/storage/netapp/na_cdot_license.py

```diff
@@ -0,0 +1,343 @@
+    choices: ['true', 'false']
+
+  remove_expired:
+    required: false
+    description:
+    - Remove licenses that have expired in the cluster.
+    choices: ['true', 'false']
+
+  serial_number:
+    required: false
+    description:
+    - Serial number of the node associated with the license. This parameter
+    - is used primarily when removing license for a specific service. If this
+    - parameter is not provided, the cluster serial number is used by default.
+
+  licenses:
+    required: false
+    description:
+    - List of licenses to add or remove.
+    
+    - Please note that trying to remove a non-existent license will throw an error.
+    valid_options:
+        - base                : Cluster Base License,
+        - nfs                 : NFS License,
+        - cifs                : CIFS License,
+        - iscsi               : iSCSI License,
+        - fcp                 : FCP License,
+        - cdmi                : CDMI License,
+        - snaprestore         : SnapRestore License,
+        - snapmirror          : SnapMirror License,
+        - flexclone           : FlexClone License,
+        - snapvault           : SnapVault License,
+        - snaplock            : SnapLock License,
+        - snapmanagersuite    : SnapManagerSuite License,
+        - snapprotectapps     : SnapProtectApp License,
+        - v_storageattach     : Virtual Attached Storage License
+
+'''
+
+
+EXAMPLES = """
+
+    - name: Add licenses
+          na_cdot_license:
+            hostname: "{{ netapp_hostname }}"
+            username: "{{ netapp_username }}"
+            password: "{{ netapp_password }}"
+            serial_number: #################
+            licenses:
+              nfs: #################
+              cifs: #################
+              iscsi: #################
+              fcp: #################
+              snaprestore: #################
+              flexclone: #################
+
+    - name: Remove licenses
+          na_cdot_license:
+            hostname: "{{ netapp_hostname }}"
+            username: "{{ netapp_username }}"
+            password: "{{ netapp_password }}"
+            remove_unused: false
+            remove_expired: true
+            serial_number: #################
+            licenses:
+              nfs: remove
+"""
+
+RETURN = """
+msg:
+    description: Successfully added licenses
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+msg:
+    description: Successfully removed licenses
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+"""
+
+import sys
+import json
+import logging
+from itertools import ifilter
+from traceback import format_exc
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
+from ansible.module_utils.pycompat24 import get_exception
+
+import socket
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+from netapp_lib.api.zapi import zapi
```

Comment:
Import custom packages in try/except and handled with fail_json() in main() e.g.::

    try:
        import foo
        HAS_LIB=True
    except:
        HAS_LIB=False